opr/assetList.go

```diff
@@ -29,6 +30,28 @@ func (o OraclePriceRecordAssetList) ContainsExactly(list []string) bool {
 	return o.Contains(list) // Check every asset in list is in ours
 }
 
+func (o OraclePriceRecordAssetList) SetValueFromUint64(asset string, value uint64) {
+	o[asset] = value
+}
+
+// SetValue converts the float to uint64
+// Deprecated: Should not be using floats, stick to the uint64
+func (o OraclePriceRecordAssetList) SetValue(asset string, value float64) {
+	// This only keeps 8 decimal places of precision
+	o[asset] = uint64(math.Round(value * 1e8))
+}
+
+// Uint64Value returns 0 for empty assets
+func (o OraclePriceRecordAssetList) Uint64Value(asset string) uint64 {
+	return o[asset]
+}
+
+// Value converts the value to a float
+// Deprecated: Should not be using floats, stick to the uint64
+func (o OraclePriceRecordAssetList) Value(asset string) float64 {
```

Comment:
Do I understand correctly that for v1 this is the value that ends up in the OPR after transformation:
```
float64(uint64(TruncateTo4Decimales(value) * 1e8)) / 1e8
```

Reply:
Yes. 

V1 stores 4 decimal place, and this is crucial. If we did not truncate it, the trailing oddities of floats would be in the uint64, and the value put in might not come out the same.

Reply:
What I meant is that in this new code you go through all those transformations `float64(uint64(TruncateTo4Decimales(value) * 1e8)) / 1e8` while in the old code we were using directly `TruncateTo4Decimales(value)` for OPR V1. What I am wondering if after all those successive transformations (i.e. floating point approximations) we always end up with the exact same result as with the old code. Otherwise it's a potential fork while V1 is still operating.

Reply:
Steven showed me a snippet of code that verifies the equality for a convincing number of examples. So it does look fine.

Reply:
https://play.golang.org/p/qyRUgPFfo41

Here is the process